### PR TITLE
Abstract method has `generics`

### PR DESCRIPTION
**Pull request by PerisicJere**

### CODE REVIEW

File: library/tests/fy_py_files/test_fy_py_abstract_method_with_generics_fy.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_abstract_method_with_generics_fy.py:1]: test

File: library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/abstract_method_fy.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/abstract_method_fy.py:1]: generated test

File: library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/abstract_method_fy.py.expected
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/abstract_method_fy.py.expected:1]: expected test

File: library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/method_greeting_t.py
Comment (PerisicJere) [inline library/tests/fy_py_files/test_fy_py_files/abstract_method_with_generics/method_greeting_t.py:1]: generic

File: library/src/fy_library/domain/fy_py_template_models.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ class MethodTemplateModel(BaseTemplateModel):
 
 class AbstractMethodTemplateModel(BaseTemplateModel):
     abstract_method_name: PythonEntityName
+    generics: str
```
Comment (vinkobuble) [inline library/src/fy_library/domain/fy_py_template_models.py:38]: `return_type_or_generics_def`

File: library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py [hunk 2]
```diff
@@ -45,14 +46,17 @@ def _abstract_method_file_split(self) -> AbstractMethodFileSplitModel:
...
         abstract_method_file_split_model = AbstractMethodFileSplitModel(
             user_imports=abstract_method_file_split[0],
             abstract_method_name=abstract_method_file_split[1],
-            arguments=abstract_method_file_split[3],
-            return_type=abstract_method_file_split[4],
+            generics=abstract_method_file_split[3]
+            if abstract_method_file_split[3] is not None
+            else "",
+            arguments=abstract_method_file_split[5],
+            return_type=abstract_method_file_split[6],
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py:59]:
```suggestion
            return_type=abstract_method_file_split[6] or abstract_method_file_split[3],
```

File: library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py [hunk 3]
```diff
@@ -45,14 +46,17 @@ def _abstract_method_file_split(self) -> AbstractMethodFileSplitModel:
         abstract_method_file_split = _ABSTRACT_METHOD_REGEX.split(self._fy_code)
 
         assert (
-            len(abstract_method_file_split) == 6
+            len(abstract_method_file_split) == 8
         ), f"Abstract Method file split length {len(abstract_method_file_split)} is invalid"
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py:51]: `assert` same as for property that either `return_type` or `generics_def` part can be defined.

**Review by vinkobuble — CHANGES_REQUESTED** [review]

File: library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py [hunk 4]
```diff
@@ -45,14 +46,17 @@ def _abstract_method_file_split(self) -> AbstractMethodFileSplitModel:
...
 
         abstract_method_file_split_model = AbstractMethodFileSplitModel(
             user_imports=abstract_method_file_split[0],
             abstract_method_name=abstract_method_file_split[1],
-            arguments=abstract_method_file_split[3],
-            return_type=abstract_method_file_split[4],
+            generics=abstract_method_file_split[3]
+            if abstract_method_file_split[3] is not None
+            else "",
```
Comment (vinkobuble) [inline library/src/fy_library/mixins/property/abstract_method_file_split/using_abstract_method_regex_fy.py:57]:
```suggestion
            generics=abstract_method_file_split[3] or "",
```

**Review by vinkobuble — CHANGES_REQUESTED** [review]

**Review by vinkobuble — APPROVED** [review]

**Pull request merged**